sanic_routing/patterns.py

```diff
@@ -7,22 +7,46 @@ def parse_date(d):
     return datetime.strptime(d, "%Y-%m-%d").date()
 
 
+def alpha(param: str) -> str:
+    if not param.isalpha():
+        raise ValueError(f"Value {param} contains non-alphabetic chracters")
+    return param
+
+
+def slug(param: str) -> str:
+    if not REGEX_TYPES["slug"][1].match(param):
+        raise ValueError(f"Value {param} does not match the slug format")
+    return param
+
+
 REGEX_PARAM_NAME = re.compile(r"^<([a-zA-Z_][a-zA-Z0-9_]*)(?::(.*))?>$")
+
+# Predefined path parameter types. The value is a tuple consisteing of a
+# callable and a compiled regular expression.
+# The callable should:
+#   1. accept a string input
+#   2. cast the string to desired type
+#   3. raise ValueError if it cannot
+# The regular expression is generally NOT used. Unless the path is forced
+# to use regex patterns.
 REGEX_TYPES = {
-    "string": (str, re.compile(r"^[^/]+")),
-    "alpha": (str, re.compile(r"^[A-Za-z]+")),
-    "path": (str, re.compile(r"^[^/]?.*?")),
-    "number": (float, re.compile(r"^-?(?:\d+(?:\.\d*)?|\.\d+)")),
-    "int": (int, re.compile(r"^-?\d+")),
+    "string": (str, re.compile(r"^[^/]+$")),
+    "str": (str, re.compile(r"^[^/]+$")),
+    "slug": (slug, re.compile(r"^[a-z0-9]+(?:-[a-z0-9]+)*$")),
+    "alpha": (alpha, re.compile(r"^[A-Za-z]+$")),
+    "path": (str, re.compile(r"^[^/]?.*?$")),
```

Comment:
Is the path regex correct? What's that trickery with optional initial non-slash?

Reply:
We have some tests for this here: https://github.com/sanic-org/sanic/blob/main/tests/test_routes.py#L582

TBH, I cannot recall the reason for making it optional. The commit is here: https://github.com/sanic-org/sanic-routing/commit/87d8adabf13a8ec6b1964f865ecf97927a615e22

It does seem like we can probably just remove it completely. Will need to do more testing though.